weather_mv/loader_pipeline/util_test.py

```diff
@@ -178,5 +185,99 @@ def test_make_attrs_ee_compatible_a2(self):
 
 
 class ToJsonSerializableTypeTests(unittest.TestCase):
-    # TODO(#106): Write tests...
-    pass
+
+    def test_to_json_serializable_type_none(self):
+        attrs = None
+
+        expected = None
+
+        actual = to_json_serializable_type(attrs)
+
+        self.assertEqual(actual, expected)
+        self.assertEqual(type(actual), type(expected))
+
+    def test_to_json_serializable_type_npfloat(self):
+        attrs = np.float32(1)
+
+        expected = float(1)
+
+        actual = to_json_serializable_type(attrs)
+
+        self.assertEqual(actual, expected)
+        self.assertEqual(type(actual), type(expected))
+
+    def test_to_json_serializable_type_ndarray(self):
+        attrs = np.arange(5)
+
+        expected = [0, 1, 2, 3, 4]
+
+        actual = to_json_serializable_type(attrs)
+
+        self.assertEqual(actual, expected)
+        self.assertEqual(type(actual), type(expected))
+
+    def test_to_json_serializable_type_ndarray_sizeone(self):
+        attrs = np.arange(1)
+
+        expected = [0]
+
+        actual = to_json_serializable_type(attrs)
+
+        self.assertEqual(actual, expected)
+        self.assertEqual(type(actual), type(expected))
+
+    def test_to_json_serializable_type_datetime(self):
+        input_date = '2000-01-01T00:00:00+00:00'
+
+        attrs = datetime.fromisoformat(input_date)
+
+        expected = input_date
+
+        actual = to_json_serializable_type(attrs)
+
+        self.assertEqual(actual, expected)
+        self.assertEqual(type(actual), type(expected))
+
+    def test_to_json_serializable_type_datetimestr(self):
+        input_date = '2000-01-01T00:00:00+00:00'
+
+        attrs = input_date
+
+        expected = input_date
+
+        actual = to_json_serializable_type(attrs)
+
+        self.assertEqual(actual, expected)
+        self.assertEqual(type(actual), type(expected))
+
+    def test_to_json_serializable_type_npdatetime(self):
+        input_date = '2000-01-01T00:00:00+00:00'
+
+        attrs = np.datetime64(input_date)
+
+        expected = input_date
+
+        actual = to_json_serializable_type(attrs)
+
+        self.assertEqual(actual, expected)
+        self.assertEqual(type(actual), type(expected))
+
+    def test_to_json_serializable_type_nptimedelta(self):
+        attrs = np.timedelta64(1, 'm')  # timedelta of 1 minute.
+
+        expected = float(60)  # seconds.
+
+        actual = to_json_serializable_type(attrs)
+
+        self.assertEqual(actual, expected)
+        self.assertEqual(type(actual), type(expected))
+
```

Comment:
Here are my internal tests that you're free to port here: 

```

  def test_time_to_isostring(self):
    now = datetime.datetime.now()
    now_str = now.isoformat()
    now_np = np.datetime64(now)

    self.assertEqual(self.convert(now), self.convert(now_str))
    self.assertEqual(self.convert(now), self.convert(now_np))

  def test_np_timedelta_to_float(self):
    delta = datetime.timedelta(seconds=33)
    delta_np = np.timedelta64(delta)
    self.assertEqual(self.convert(delta_np), delta.total_seconds())

  def test_np_integer_to_int(self):
    self.assertEqual(self.convert(np.int64(1)), 1)
    self.assertEqual(self.convert(np.int64(-10_000)), -10_000)
    self.assertEqual(self.convert(np.uint16(25)), 25)
```

Reply:
Thanks.